### PR TITLE
feat(destination): onboard awin integration

### PR DESCRIPTION
**Pull request by rohithkaza**
## Description of the change

> Onboarding Awin Cloud Mode Destination

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

File: v0/destinations/awin/config.js [hunk 1]
```diff
@@ -0,0 +1,8 @@
+const BASE_URL = "https://www.awin1.com/sread.php?tt=ss&tv=2";
```
Comment (ItsSudip) [inline v0/destinations/awin/config.js:1]:
```suggestion
const BASE_URL = "https://www.awin1.com/sread.php";
```
We shouldn't provide the query params here.

File: v0/destinations/awin/transform.js [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+  checkInvalidRtTfEvents
+} = require("../../util");
+
+const { getEndpoint } = require("./utils");
+
+const responseBuilder = (message, { Config }) => {
+  const { advertiserId, eventsToTrack } = Config;
+  let endpoint = BASE_URL.concat("&merchant=", advertiserId);
+  // if the event is present in eventsToTrack List
+  if (eventsToTrack.includes(message.event)) {
```
Comment (ItsSudip) [inline v0/destinations/awin/transform.js:19]: We should check if `eventsToTrack` is an array or not.

File: v0/destinations/awin/utils.js [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+  }
+};
+
+const generateEndpoint = (endpoint, paramsIds, paramsValues) => {
+  for (let i = 0; i < paramsIds.length; i++) {
+    if (isDefinedAndNotNull(paramsValues[i])) {
+      let paramsValue = String(paramsValues[i]);
+      if (paramsIds[i] === "parts") {
+        if (isDefinedAndNotNull(paramsValues[0])) {
+          endpoint = endpoint.concat(`&${paramsIds[i]}=${paramsValue}`);
```
Comment (ItsSudip) [inline v0/destinations/awin/utils.js:52]: Do we need to generate the endpoint in this format? We can put them inside `defaultRequestConfig().params`.

File: v0/destinations/awin/utils.js [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+      properties.orderReference ||
+      properties.order_reference;
+    testMode =
+      properties.testMode ||
+      properties.test_mode ||
+      properties.isTest ||
+      properties.is_test ||
+      0;
+    cks = properties.cks || properties.awc || "awc";
+  }
```
Comment (ItsSudip) [inline v0/destinations/awin/utils.js:43]: We can use constructPayload function here.

**Review by ItsSudip — CHANGES_REQUESTED** [review]

**Comment by sandeepdsvs** [comment]
@rohithkaza please make sure you clear the issues detected by DeepSource in the below checks

File: v0/destinations/awin/config.js (hunk 1 above)
Comment (rohithkaza) [reply v0/destinations/awin/config.js]: Removed

File: v0/destinations/awin/transform.js (hunk 2 above)
Comment (rohithkaza) [reply v0/destinations/awin/transform.js]: Created a new Array and used it

File: v0/destinations/awin/utils.js (hunk 3 above)
Comment (rohithkaza) [reply v0/destinations/awin/utils.js]: Done

File: v0/destinations/awin/utils.js (hunk 4 above)
Comment (rohithkaza) [reply v0/destinations/awin/utils.js:45]: Not sure, why we need constructPayload function here.
Do you mean to keep these things in mappingConfig and store them in the payload to retrieve the params object directly(to use it later)?

Comment (sandeepdsvs) [reply v0/destinations/awin/utils.js:49]: > Do you mean to keep these things in mappingConfig

Yes. Then you can use constructPayload and get the params object and use it directly in generateParams function. This is a better approach the current one (setting global variables and then use them later)

File: v0/destinations/awin/transform.js (hunk 2 above)
Comment (ItsSudip) [reply v0/destinations/awin/transform.js]: What if `eventsToTrack` is not an array?

Comment (rohithkaza) [reply v0/destinations/awin/transform.js]: `eventsToTrack` is a dynamicCustomForm, which is in the format of an Array of Objects[ {...}, {...}, ...].
Anyway, I can add that validation.

File: v0/destinations/awin/utils.js (hunk 4 above)
Comment (rohithkaza) [reply v0/destinations/awin/utils.js]: Done

File: v0/destinations/awin/utils.js [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+      params.commissionGroup = "DEFAULT";
+    }
+    params.parts = String(params.commissionGroup).concat(
+      ":",
+      String(params.amount)
+    );
+    delete params.commissionGroup;
+  }
+  if (!params.cks) {
+    params.cks = "awc";
```
Comment (ItsSudip) [inline v0/destinations/awin/utils.js:23]: Why are we setting it to some default value?

File: v0/destinations/awin/data/trackConfig.json [hunk 6]
```diff
@@ -0,0 +1,59 @@
+[
+  {
+    "destKey": "params.amount",
+    "sourceKeys": [
+      "properties.revenue",
+      "properties.amount",
+      "properties.totalAmount",
+      "properties.total_amount"
```
Comment (ItsSudip) [inline v0/destinations/awin/data/trackConfig.json:8]:
```suggestion
      "properties.amount",
      "properties.revenue",
      "properties.total",
      "properties.totalAmount",
      "properties.total_amount"
```

**Review by ItsSudip — CHANGES_REQUESTED** [review]

File: v0/destinations/awin/utils.js (hunk 5 above)
Comment (rohithkaza) [reply v0/destinations/awin/utils.js]: Removed the default value

File: v0/destinations/awin/data/trackConfig.json (hunk 6 above)
Comment (rohithkaza) [reply v0/destinations/awin/data/trackConfig.json]: Done

**Review by ItsSudip — APPROVED** [review]

**Review by sandeepdsvs — APPROVED** [review]

**Pull request merged**